src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala

```diff
@@ -225,9 +225,21 @@ object DiplomaticObjectModelAddressing {
       grandParentOpt = b.device.get.parent
       gp <- grandParentOpt
     } yield OMInterrupt(
-            receiver = getDeviceName(gp),
-            numberAtReceiver = getInterruptNumber(b.value),
-            name = name
-          )
-    }
-}
+      receiver = getDeviceName(gp, resources),
+      numberAtReceiver = getInterruptNumber(b.value),
+      name = name
+    )
+  }
+
+  def describeGlobalInterrupts(name: String, resources: ResourceBindings): Seq[OMInterrupt] = {
+    val bindings = resources("int")
+    for {
+      binding <- bindings
+      device = binding.device.get
```

Comment:
I don't follow why `describeInterrupts` uses `device.get.parent`, but  `descibeGlobalInterrupts` stops at `device.get`. We need the parent device in all cases I can think of. The same also makes the `receiver` property inconsistent.

Reply:
I mentioned that the Device that these interrupts are extracted from is not the same type of device as the PLIC device which gets the parent. I do not see a way to get the target (PLIC) device using the Interrupts device.